cdk/domino_cdk/eks_stack.py

```diff
@@ -425,20 +488,23 @@ def provision_managed_nodegroup(self, name: str, cfg: dict, max_nodegroup_azs: i
                         )
                     ],
                     machine_image=machine_image,
-                    user_data=ec2.UserData.custom(cdk.Fn.sub(user_data, {"ClusterName": self.cluster.cluster_name})),
+                    user_data=mime_user_data,
                 )
                 lts = eks.LaunchTemplateSpec(id=lt.launch_template_id, version=lt.version_number)
                 disk_size = None
+            else:
+                disk_size = cfg["disk_size"]
+                lts = None
 
             key_name = cfg.get("key_name", None)
             self.cluster.add_nodegroup_capacity(
-                f"{name}-{i}",  # this might be dangerous
-                nodegroup_name=f"{self.name}-{name}-{az}",  # this might be dangerous
+                f"{self.name}-{name}-{i}",  # this might be dangerous
+                nodegroup_name=f"{self.name}-{name}-{i}",
```

Comment:
ok

Reply:
unfortunately, cannot use az for id because az is a token.

I have changed for the name, though

See https://us-west-2.console.aws.amazon.com/ec2autoscaling/home?region=us-west-2#/details/denis-test1-nvidia-us-west-2a?view=details

Also, please be aware, that for the managed groups EKS takes the provided group as a template and creates a new group with the same parameters but random ID. I believe this is part of managing, it clones what we have provided and changes it to the current policies.

See https://us-west-2.console.aws.amazon.com/ec2autoscaling/home?region=us-west-2#/details/eks-dabcff72-4cc4-d1c4-8d6d-73dfb6b5d904?view=details